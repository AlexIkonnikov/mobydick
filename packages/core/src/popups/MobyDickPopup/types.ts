import {FC} from 'react';

import {IPopupsContext} from '../context';
import {IContentProps, IPopup} from '../types';

export type ModalProps = Partial<Omit<IContentProps, 'onClose'>> & {
  onClose?(id: string): void;
};

export interface IPopupWithProps extends IPopup {
  props: Record<string, unknown> & ModalProps;
}

export interface IModalRef extends Pick<IPopupsContext, 'closeAllPopups'> {
  openPopup<Props>(
    Component: FC<Props & IContentProps>,
    props?: Props & ModalProps,
  ): void;
  closePopup(id: string): void;
}
