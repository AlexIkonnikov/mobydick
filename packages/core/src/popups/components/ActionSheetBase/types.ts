import {StyleProp, ViewStyle} from 'react-native';
import React, {PropsWithChildren, ReactElement} from 'react';

import {TypographyProp} from '../../../typography/types';
import {IPopup} from '../../types';

export interface IPropsContents {
  title: string;
  textFont?: TypographyProp;
  leftIcon?: ReactElement | undefined;
  onPress?(): void;
  radio?: string;
  checkboxList?: string[];
  disabled?: boolean;
  itemType: IItemType;
}

export enum IItemType {
  firstItem = 'firstItem',
  innerItem = 'innerItem',
  lastItem = 'lastItem',
  singleItem = 'singleItem',
  cancelItem = 'cancelItem',
}

export interface IPropsItem extends IPropsContents {
  style?: ViewStyle;
  isStatusPressedForTest?: boolean;
}

export interface IActionSheetBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  children: React.ReactNode;
}
