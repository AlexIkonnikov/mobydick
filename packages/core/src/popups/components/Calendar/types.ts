export interface IChangeDate {
  dateStart: string;
  dateEnd: string;
}

export enum IButtonView {
  small = 'small',
  large = 'large',
}
