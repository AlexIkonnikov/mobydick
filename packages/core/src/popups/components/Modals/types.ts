import {IButtonTypes} from '../../../cta';

export interface IModalProps {
  title: string;
  descriptionText: string;
  buttonText?: string;
}

export interface IModalAskProps {
  title: string;
  descriptionText: string;

  typeLeft?: IButtonTypes;
  textLeft?: string;

  onPressRight(): void;
  typeRight?: IButtonTypes;
  textRight?: string;
}
