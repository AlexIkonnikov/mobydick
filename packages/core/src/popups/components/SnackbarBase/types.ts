import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren} from 'react';

import {IPopup, IPosition} from '../../types';

export interface ISnackbarProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  containerStyle?: StyleProp<ViewStyle>;
  position: IPosition;
  timeShow?: number;
}
