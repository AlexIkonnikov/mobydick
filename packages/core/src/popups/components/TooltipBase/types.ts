import {StyleProp, ViewStyle} from 'react-native';
import {PropsWithChildren, RefObject} from 'react';

import {IPopup, IPosition} from '../../types';
import {ITouchableOpacity} from '../../../basic';

export enum IPlacement {
  start = 'start',
  center = 'center',
  end = 'end',
}

export interface IArrowViewPopup {
  placement: IPlacement;
  position: IPosition;
  arrowViewStyles?: ViewStyle | ViewStyle[];
}

export interface ITooltipBaseProps
  extends PropsWithChildren<Omit<IPopup, 'Content'>> {
  onClose: () => void;
  position: IPosition;
  placement: IPlacement;
  refCurrent: RefObject<ITouchableOpacity>;
  overlayStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  timeShow?: number;
}
