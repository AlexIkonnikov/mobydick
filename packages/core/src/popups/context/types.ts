import {IPopup, IPopupId} from '../types';

export type IOpenPopupParams = Partial<IPopup> & Pick<IPopup, 'Content'>;

export interface IPopupsContext {
  popups: IPopup[];
  openPopup: (popup: IOpenPopupParams) => string;
  closePopup: (id: IPopupId) => void;
  closeAllPopups: () => void;
}
