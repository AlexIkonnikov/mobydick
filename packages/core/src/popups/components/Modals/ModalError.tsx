import React, {FC} from 'react';

import {ModalBase} from '../ModalBase';
import {IContentProps} from '../../types';
import {IButtonTypes} from '../../../cta/components/Button/types';
import useTheme from '../../../styles/theme/hooks/useTheme';
import useStyles from '../../../styles/theme/hooks/useStyles';

import stylesCreate from './stylesCreate';

interface IProps {
  title: string;
  descriptionText: string;

  typeLeft?: IButtonTypes;
  textLeft?: string;

  onPressRight(): void;
  typeRight?: IButtonTypes;
  textRight?: string;
}

const ModalError: FC<IContentProps & IProps> = props => {
  const {
    title,
    descriptionText,
    typeLeft,
    textLeft,
    typeRight,
    textRight,
    onPressRight,
    onClose,
  } = props;
  const {colors} = useTheme();
  const [styles] = useStyles(stylesCreate);

  return (
    <ModalBase
      overlayStyle={styles.overlayStyle}
      containerStyle={styles.container}
      {...props}>
      <ModalBase.AlertContent
        name={'icon-warning'}
        color={colors.IconAttention}
        style={{backgroundColor: colors.BgError}}
      />

      <ModalBase.TextContent title={title} descriptionText={descriptionText} />
      <ModalBase.HorizontalButtonsView
        typeLeft={typeLeft || IButtonTypes.tertiary}
        textLeft={textLeft || 'Отмена'}
        onPressLeft={onClose}
        typeRight={typeRight || IButtonTypes.destructive}
        textRight={textRight || 'Удалить'}
        onPressRight={onPressRight}
      />
    </ModalBase>
  );
};

export default ModalError;
