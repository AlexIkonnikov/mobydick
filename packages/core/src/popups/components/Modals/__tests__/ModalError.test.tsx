import {render} from '@testing-library/react-native';
import React from 'react';

import {IButtonTypes} from '../../../../cta/components/Button/types';
import ModalError from '../ModalError';

describe('@npm/mobydick-core/modalError', () => {
  it('should renders correctly', () => {
    const {toJSON} = render(
      <ModalError
        title={'title'}
        descriptionText={'descriptionText'}
        onPressRight={() => null}
        id={'id'}
        onClose={() => null}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with optional', () => {
    const {toJSON} = render(
      <ModalError
        title={'title'}
        descriptionText={'descriptionText'}
        onPressRight={() => null}
        typeRight={IButtonTypes.primary}
        textRight={'textRight'}
        typeLeft={IButtonTypes.primary}
        textLeft={'textLeft'}
        id={'id'}
        onClose={() => null}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
