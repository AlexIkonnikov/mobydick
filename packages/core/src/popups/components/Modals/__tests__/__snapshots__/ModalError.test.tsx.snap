// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-core/modalError should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      [
        {
          "justifyContent": "flex-end",
        },
        {
          "justifyContent": "center",
        },
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 36,
          "justifyContent": "center",
          "marginVertical": 60,
          "paddingHorizontal": 60,
          "paddingVertical": 72,
          "width": 702,
        },
        {
          "maxWidth": "80%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(43, 120, 238, 0.15)",
            "borderRadius": 100,
            "padding": 36,
          },
          {
            "backgroundColor": "rgba(245, 77, 61, 0.15)",
          },
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          {
            "color": "#F54D3D",
            "fontFamily": "Neotis",
            "fontSize": 108,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        {
          "marginBottom": 24,
          "marginTop": 36,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#20242D",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 54,
              "lineHeight": 66,
            },
            [
              {
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        title
      </Text>
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#5E6678",
              "fontFamily": "Inter-Regular",
              "fontSize": 42,
              "lineHeight": 54,
            },
            [
              {
                "paddingTop": 24,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        descriptionText
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "paddingTop": 60,
        }
      }
    >
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "transparent",
            "borderRadius": 36,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginRight": 18,
            "minHeight": 144,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 54,
                "lineHeight": 66,
              },
              [
                {
                  "color": "#2B78EE",
                  "paddingHorizontal": 18,
                },
                undefined,
              ],
            ]
          }
        >
          Отмена
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#F54D3D",
            "borderRadius": 36,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 18,
            "minHeight": 144,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 54,
                "lineHeight": 66,
              },
              [
                {
                  "color": "#FFF",
                  "paddingHorizontal": 18,
                },
                undefined,
              ],
            ]
          }
        >
          Удалить
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`@npm/mobydick-core/modalError should renders correctly with optional 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.2)",
        "height": "100%",
        "position": "absolute",
        "width": "100%",
      },
      [
        {
          "justifyContent": "flex-end",
        },
        {
          "justifyContent": "center",
        },
      ],
    ]
  }
  testID="Pressable"
>
  <View
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 36,
          "justifyContent": "center",
          "marginVertical": 60,
          "paddingHorizontal": 60,
          "paddingVertical": 72,
          "width": 702,
        },
        {
          "maxWidth": "80%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(43, 120, 238, 0.15)",
            "borderRadius": 100,
            "padding": 36,
          },
          {
            "backgroundColor": "rgba(245, 77, 61, 0.15)",
          },
        ]
      }
    >
      <Text
        allowFontScaling={true}
        style={
          {
            "color": "#F54D3D",
            "fontFamily": "Neotis",
            "fontSize": 108,
          }
        }
      >
        
      </Text>
    </View>
    <View
      style={
        {
          "marginBottom": 24,
          "marginTop": 36,
        }
      }
    >
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#20242D",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 54,
              "lineHeight": 66,
            },
            [
              {
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        title
      </Text>
      <Text
        allowFontScaling={true}
        style={
          [
            {
              "color": "#5E6678",
              "fontFamily": "Inter-Regular",
              "fontSize": 42,
              "lineHeight": 54,
            },
            [
              {
                "paddingTop": 24,
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
      >
        descriptionText
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "paddingTop": 60,
        }
      }
    >
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#2B78EE",
            "borderRadius": 36,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginRight": 18,
            "minHeight": 144,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 54,
                "lineHeight": 66,
              },
              [
                {
                  "color": "#FFF",
                  "paddingHorizontal": 18,
                },
                undefined,
              ],
            ]
          }
        >
          textLeft
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#2B78EE",
            "borderRadius": 36,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 18,
            "minHeight": 144,
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={true}
          style={
            [
              {
                "color": "#FFF",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 54,
                "lineHeight": 66,
              },
              [
                {
                  "color": "#FFF",
                  "paddingHorizontal": 18,
                },
                undefined,
              ],
            ]
          }
        >
          textRight
        </Text>
      </View>
    </View>
  </View>
</View>
`;
